business/proxy_logging.go

```diff
@@ -0,0 +1,46 @@
+package business
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/util/httputil"
+)
+
+const (
+	envoyAdminPort = 15000
+)
+
+type ProxyLoggingService struct {
+	k8s kubernetes.ClientInterface
+}
+
+func (in *ProxyLoggingService) SetLogLevel(ctx context.Context, namespace, pod, level string) error {
+	path := fmt.Sprintf("/logging?level=%s", level)
+
+	localPort := httputil.Pool.GetFreePort()
+	f, err := in.k8s.GetPodPortForwarder(namespace, pod, fmt.Sprintf("%d:%d", localPort, envoyAdminPort))
```

Comment:
Does this need to return a pointer to an `Interface`? Can it be updated to just return a `PortForwarder` interface? Then start can be called as `f.Start()` instead of `(*f).Start()`.

Reply:
I guess these wraps are more inspired from the client itself
![image](https://user-images.githubusercontent.com/1662329/132377942-cc4fff43-92fd-43f9-a4af-30502bbea9dd.png)

That they use the pointer to interface to make the type opaque and just focused on the interface.

This is more code style, as @xeviknal was the original author, when he's back that design can be revisited, but looking the client-go I see some similar interfaces, I guess that's the rationale.

Reply:
Yeah it's definitely just code style. I just thought it looked odd when calling/mocking that package but it's pretty unimportant . The difference between the `PortForwarder` in client-go and the `PortForwarder` in kiali is that in client-go it is a [struct](https://pkg.go.dev/k8s.io/client-go@v0.22.1/tools/portforward#PortForwarder) whereas in kiali it is an [interface](https://pkg.go.dev/github.com/kiali/kiali@v1.39.0/util/httputil#PortForwarder).